examples/media/multipart/src/main/java/io/helidon/examples/media/multipart/FileService.java

```diff
@@ -125,41 +107,22 @@ private void streamUpload(ServerRequest req, ServerResponse res) {
                     res.send();
                 }).forEach((part) -> {
                     if ("file[]".equals(part.name())) {
-                        final ByteChannel channel = newByteChannel(storage, part.filename());
-                        Multi.create(part.content())
+                        final ByteChannel channel = newByteChannel(storage.create(part.filename()));
+                        part.content()
                                 .forEach(chunk -> writeChunk(channel, chunk))
                                 .thenAccept(it -> closeChannel(channel));
                     }
                 });
     }
 
-    private static Path createStorage() {
-        try {
-            return Files.createTempDirectory("fileupload");
-        } catch (IOException ex) {
-            throw new RuntimeException(ex);
-        }
-    }
-
-    private static Stream<String> listFiles(Path storage) {
-        try {
-            return Files.walk(storage)
-                    .filter(Files::isRegularFile)
-                    .map(storage::relativize)
-                    .map(Path::toString);
-        } catch (IOException ex) {
-            throw new RuntimeException(ex);
-        }
-    }
-
-    private static void writeBytes(Path storage, String fname, byte[] bytes) {
+    private static void writeBytes(Path file, byte[] bytes) {
         try {
-            Files.write(storage.resolve(fname), bytes,
+            Files.write(file, bytes,
```

Comment:
This is a blocking operation that is not allowed within a reactive server processing.

Reply:
The only alternative is to create a dedicated thread-pool, shall I do that  ?

Reply:
Please create a follow up issue - we need to resolve this for other examples as well.

Reply:
Yes and there is also the blocking start of the webserver in the unit test, that you pointed out in the new multiport PR. I copied that from the quickstarts, so I will create another issue to address that.

Reply:
created #2844 and #2845